File: Models/Payment.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Delivora.Models;

public class Payment
{
    public int Id { get; set; }

    public decimal Amount { get; set; } = 0;

    public PaymentStatus Status { get; set; }

    public DateTime PaymentDate { get; set; } = DateTime.Now;


    // Relationships
    public int OrderId { get; set; }

    [ForeignKey(nameof(OrderId))]
    public Order Order { get; set; } = default!;



    public int PaymentMethodId { get; set; }

    [ForeignKey(nameof(PaymentMethodId))]
    public PaymentMethod PaymentMethod { get; set; } = default!;

}

File: angular-frontend/src/app/core/models/payment/payment.ts



export interface Payment {
    id: number;
    amount: number;
    status: string;
    paymentDate: string;
    orderId: number;
    methodName: string;
}



export enum PaymentStatus {
    Pending = 0,
    Paid = 1,
    Failed = 2,
    Refunded = 3
}